rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/interpreter/SymbolTableTest.kt

```diff
@@ -0,0 +1,152 @@
+package com.smeup.rpgparser.interpreter
+
+import com.smeup.rpgparser.AbstractTest
+import com.smeup.rpgparser.PerformanceTest
+import com.smeup.rpgparser.parsing.ast.CompilationUnit
+import org.junit.experimental.categories.Category
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertIs
+import kotlin.test.assertNull
+import kotlin.time.measureTime
+
+/**
+ * The purpose of this test suite is to validate the behaviour around Symbol Table.
+ * The resources about this suite starts with `ST` (Symbol Table), followed by any string which describes the purpose.
+ */
+class SymbolTableTest : AbstractTest() {
+    /**
+     * Performance test for accessing standalone fields and data structure fields in a symbol table.
+     *
+     * This test measures the execution time required to perform repeated lookups of standalone fields and
+     * data structure fields in a symbol table created from an Abstract Syntax Tree (AST). It verifies that
+     * the AST can be produced successfully and evaluates the performance of symbol table lookups for
+     * specific fields.
+     *
+     * Steps:
+     * 1. Produce an AST from the specified example program (`symboltable/ST_PERFORMANCE_ACCESS01`).
+     * 2. Create a symbol table (`ISymbolTable`) from the AST.
+     * 3. Perform 1,000,000 lookups for the standalone field `VAR1`.
+     * 4. Perform 1,000,000 lookups for the last field of the data structure `DS10_FLD50`.
+     * 5. Measure and log the total execution time for these operations.
+     *
+     * The goal is to evaluate the efficiency of symbol table lookups and ensure performance is within an acceptable range.
+     *
+     * @throws TimeoutException if the test does not complete within 6 seconds
+     * @see ISymbolTable
+     */
+    @Test(timeout = 3_000)
+    @Category(PerformanceTest::class)
+    fun executeST_F_WITH_DS_UNQUALIFIED1_PERFORMANCE() {
+        assertASTCanBeProduced(
```

Comment:
With this approach you are testing also the ast loading, but we want to test the performances just for the `dataDefinitionByName` function.
Remove timeout from `@Test` annotation and handle this constraint after measured the total time.